internal/controller/componentrelease/controller.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2025 The OpenChoreo Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package componentrelease
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	openchoreov1alpha1 "github.com/openchoreo/openchoreo/api/v1alpha1"
+)
+
+// Reconciler reconciles a ComponentRelease object
+type Reconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=openchoreo.dev,resources=componentreleases,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=openchoreo.dev,resources=componentreleases/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=openchoreo.dev,resources=componentreleases/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// No-op controller - ComponentEnvSnapshots are managed externally
+	return ctrl.Result{}, nil
+}
```

Comment:
if reconciliation is not needed here, why do we need to introduce this controller?

Reply:
This is the same pattern we followed. I think technically, it is possible to remove the controller. @Mirage20 wdyt?

Reply:
CRDs can exist without a controller without any issues - if it’s not doing anything no point having it

Reply:
We might need the controller here, as we need to implement a finalizer logic for ComponentRelease